zerver/lib/validator.py

```diff
@@ -228,7 +228,22 @@ def check_url(var_name: str, val: object) -> Optional[str]:
     except ValidationError:
         return _('%s is not a URL') % (var_name,)
 
-def validate_field_data(field_data: ProfileFieldData) -> Optional[str]:
+def check_url_pattern(var_name: str, val: object) -> Optional[str]:
+    error = check_string(var_name, val)
+    if error:
+        return error
+    val = cast(str, val)
+
+    if val.count('%(username)s') != 1:
+        return _('username should appear exactly once in pattern.')
+    val.replace('username', '')
```

Comment:
This should be `val.replace("%(username)s, "username")`, I think

Reply:
Also, I think you want to make a new variable name -- otherwise you're potentially mutating the argument, which you don't want.